go-controller/pkg/ovn/master.go

```diff
@@ -670,7 +671,11 @@ func (oc *Controller) syncGatewayLogicalNetwork(node *kapi.Node, l3GatewayConfig
 		if err != nil {
 			return err
 		}
-		if err := addPolicyBasedRoutes(node.Name, hostIfAddr.IP.String(), l3GatewayConfigIP); err != nil {
+		relevantHostIPs, err := util.MatchAllIPStringFamily(utilnet.IsIPv6(hostIfAddr.IP), hostAddrs.List())
+		if err != nil {
```

Comment:
Doesn't this risk breaking us? `MatchAllIPStringFamily` returns an error if no IP addresses are returned...could there be situations where that is the case? We won't proceed to `addPolicyBasedRoutes` then. I guess we should always find the host's IP though, so maybe not?

Reply:
Actually absolutely, this breaks local gateway mode where we never even run the `node_ip_handler.go` which results in this always being empty.

Reply:
how did it pass the local gw mode jobs?

Reply:
i'll fix it either way